server/src/main/java/de/uftos/builders/SpecificationBuilder.java

```diff
@@ -0,0 +1,41 @@
+package de.uftos.builders;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.domain.Specification;
+
+public class SpecificationBuilder<T> {
+
+  private Specification<T> specification;
+
+  public SpecificationBuilder() {
+    specification = Specification.where(null);
+  }
+
+  public Specification<T> build() {
+    return specification;
+  }
+
+  public SpecificationBuilder<T> addOptionalOr(Optional<String> param, String paramName) {
+    if (param.isEmpty()) return this;
+    specification = specification.or(createOrFilter(param.get(), paramName));
+    return this;
+  }
+
+  public SpecificationBuilder<T> addOptionalJoinFilter(Optional<String[]> attributeValue, String relationName, String attributeName) {
```

Comment:
This one and the method above imply (by their name) that they are very generic, while in reality they specifically do LIKE and IN. Certainly there will be more methods necessary in the future, so it may be worth considering how to make those more generic

Reply:
Not knowing what our feature needs will be, I think it would be quite hard to generalize this method right now.

Reply:
Then we should probably name it less generic.

Reply:
done